src/include/parser/sql_statement.h

```diff
@@ -215,5 +218,335 @@ class TableRefStatement : public SQLStatement {
   std::unique_ptr<TableInfo> table_info_ = nullptr;
 };
 
+/**
+ * ColumnDefinition represents the logical description of a table column.
+ */
+struct ColumnDefinition {
+  // TODO(WAN): I really hate how everything is mashed together.
+  // There were also a number of unused attributes e.g. primary_keys, multi_unique...
+  // that were never used.
+
+  /**
+   * Column data types.
+   */
+  enum class DataType {
+    INVALID,
+
+    PRIMARY,
+    FOREIGN,
+    MULTIUNIQUE,
+
+    CHAR,
+    INT,
+    INTEGER,
+    TINYINT,
+    SMALLINT,
+    BIGINT,
+    DOUBLE,
+    FLOAT,
+    DECIMAL,
+    BOOLEAN,
+    ADDRESS,
+    DATE,
+    TIMESTAMP,
+    TEXT,
+
+    VARCHAR,
+    VARBINARY
+  };
+
+  /**
+   * Foreign key constructor.
+   * @param fk_sources foreign key sources
+   * @param fk_sinks foreign key sinks
+   * @param fk_sink_table_name foreign key sink table name
+   * @param delete_action action to take upon deletion
+   * @param update_action action to take upon update
+   * @param match_type type of foreign key match
+   */
+  ColumnDefinition(std::vector<std::string> fk_sources, std::vector<std::string> fk_sinks,
+                   std::string fk_sink_table_name, FKConstrActionType delete_action, FKConstrActionType update_action,
+                   FKConstrMatchType match_type)
+      : type_(DataType::FOREIGN),
+        fk_sources_(std::move(fk_sources)),
+        fk_sinks_(std::move(fk_sinks)),
+        fk_sink_table_name_(std::move(fk_sink_table_name)),
+        fk_delete_action_(delete_action),
+        fk_update_action_(update_action),
+        fk_match_type_(match_type) {}
+
+  /**
+   * Table column constructor.
+   * @param name column name
+   * @param type column type
+   * @param is_primary is primary key
+   * @param is_not_null is not nullable
+   * @param is_unique is unique
+   * @param default_expr default expression
+   * @param check_expr check expression
+   * @param varlen size of column if varlen
+   */
+  ColumnDefinition(std::string name, DataType type, bool is_primary, bool is_not_null, bool is_unique,
```

Comment:
Shouldn't parameters be const?

Reply:
const is pointless when are passing arguments as values